Client/mods/deathmatch/logic/CClientPed.cpp

```diff
@@ -6376,3 +6403,628 @@ void CClientPed::UpdateStreamPosition(const CVector& vecInPosition)
+        // is the driver's door. Force an enter for the passenger seat.
+        uiSeat = 1;
+    }
+    else if (!bPassenger)
+    {
+        // We want to drive. Force our seat to the driver's seat.
+        uiSeat = 0;
+    }
+
+    // If the vehicle's a boat, make sure we're standing on it (we need a dif task to enter boats properly)
+    if (pVehicle->GetVehicleType() == CLIENTVEHICLE_BOAT && GetContactEntity() != pVehicle)
+    {
+        return false;
+    }
+
+    // Call the onClientVehicleStartEnter event for the ped
+    // Check if it is cancelled before sending packet
+    CLuaArguments Arguments;
+    Arguments.PushElement(this);             // player / ped
+    Arguments.PushNumber(uiSeat);            // seat
+    Arguments.PushNumber(uiDoor);            // door
+
+    if (!pVehicle->CallEvent("onClientVehicleStartEnter", Arguments, true))
+    {
+        // Event has been cancelled
+        return false;
+    }
+
+    // Send an in request
+    NetBitStreamInterface* pBitStream = g_pNet->AllocateNetBitStream();
+    if (!pBitStream)
+    {
+        return false;
+    }
+
+    // Write the ped ID to it if server supports it
+    if (g_pNet->CanServerBitStream(eBitStreamVersion::PedEnterExit))
+    {
+        pBitStream->Write(GetID());
+    }
+
+    // Write the vehicle id to it and that we're requesting to get into it
+    pBitStream->Write(pVehicle->GetID());
+    unsigned char ucAction = static_cast<unsigned char>(CClientGame::VEHICLE_REQUEST_IN);
+    unsigned char ucSeat = static_cast<unsigned char>(uiSeat);
+    bool          bIsOnWater = pVehicle->IsOnWater();
+    unsigned char ucDoor = static_cast<unsigned char>(uiDoor);
+    pBitStream->WriteBits(&ucAction, 4);
+    pBitStream->WriteBits(&ucSeat, 4);
+    pBitStream->WriteBit(bIsOnWater);
+    pBitStream->WriteBits(&ucDoor, 3);
+
+    // Send and destroy it
+    g_pNet->SendPacket(PACKET_ID_VEHICLE_INOUT, pBitStream, PACKET_PRIORITY_HIGH, PACKET_RELIABILITY_RELIABLE_ORDERED);
+    g_pNet->DeallocateNetBitStream(pBitStream);
+
+    // We're now entering a vehicle
+    m_bIsGettingIntoVehicle = true;
+    m_ulLastVehicleInOutTime = CClientTime::GetTime();
+
+#ifdef MTA_DEBUG
+    g_pCore->GetConsole()->Printf("* Sent_InOut: vehicle_request_in");
+#endif
+
+    return true;
+}
+
+//////////////////////////////////////////////////////////////////
+//
+// CClientPed::ExitVehicle
+//
+// Asks server for permission to start exiting vehicle
+//
+//////////////////////////////////////////////////////////////////
+bool CClientPed::ExitVehicle()
+{
+    // Are we local player or ped we are syncing
+    if (!IsSyncing() && !IsLocalPlayer())
+    {
+        return false;
+    }
+
+    // Are we a clientside ped
+    // TODO: Add support for clientside peds
+    if (IsLocalEntity())
+    {
+        return false;
+    }
+
+    // Get our occupied vehicle
+    CClientVehicle* pOccupiedVehicle = GetOccupiedVehicle();
+    if (!pOccupiedVehicle)
+    {
+        return false;
+    }
```

Comment:
Minor detail, but this confuses me.
I think you should use a dassert here to indicate that this shoudn't happen at all, since why would `ExitVehicle` be called if you aren't in a veh in the first place?

Reply:
This function is called from CLuaPedDefs

Reply:
Yeah, my bad.